### PR TITLE
Bar graph side by side

### PR DESCRIPTION
**Pull request by drew-thorson**

### CODE REVIEW

File: src/app/d3components/bargraph/BarGraph.js [hunk 1]
```diff
@@ -51,20 +52,35 @@ class BarGraph extends React.Component {
...
 
+  stackType() {
+    let tempArray = [];
+    const d = this.stacked;
+    for (let i in d) {
+      for (let j in d[i]) {
+        if (this.props.barChartType === "side") {
+          tempArray.push(d[i][j].y);
+        } else {
+          tempArray.push(d[i][j].y + d[i][j].y0);
```
Comment (coryhouse) [inline src/app/d3components/bargraph/BarGraph.js]: This is quite cryptic. Is there a way to label these variables more descriptively? Seeing a lot of single character variables throughout.

File: src/app/d3components/bargraph/BarGraph.js [hunk 2]
```diff
@@ -123,58 +141,71 @@ class BarGraph extends React.Component {
...
+    this.props.keys.forEach((value, i) => {
+      const legendObj = {};
+      legendObj[_self.props.labelKey] = value;
+      legend[i] = legendObj;
+    });
+
     return (
       <div>
-        {title}
+        {this.props.title ? <h3>{this.props.title}</h3> : null}
```
Comment (coryhouse) [inline src/app/d3components/bargraph/BarGraph.js]: If preferred, can use this pattern:

`{this.props.title && <h3>{this.props.title}</h3>}`

Then the H3 title only displays when there is a title.

File: src/app/d3components/bargraph/BarGraph.js [hunk 3]
```diff
@@ -123,58 +141,71 @@ class BarGraph extends React.Component {
...
+              <AxisLabel key={0} h={this.h} w={this.w} axisLabel={this.props.xAxisLabel} axisType="x" />
+            : null}
+            {this.props.yAxisLabel ?
+              <AxisLabel key={1} h={this.h} w={this.w} axisLabel={this.props.yAxisLabel} axisType="y" />
+            : null}
             {bars}
           </g>
         </svg>
+        {this.props.legend ?
+        <div>
```
Comment (coryhouse) [inline src/app/d3components/bargraph/BarGraph.js]: If the legend needs a div, consider placing the wrapping div within the legend.

**Comment by coryhouse** [comment]
I added a few notes, but otherwise looks good. 👍

File: src/app/d3components/bargraph/BarGraph.js (hunk 1 above)
Comment (drew-thorson) [inline src/app/d3components/bargraph/BarGraph.js]: These variables are just for incrementing. d stands for the data from this.stacked. i and j are just indexes.

**Pull request merged**